src/main/java/upb/dice/twitter/IDHandler.java

```diff
@@ -0,0 +1,131 @@
+package upb.dice.twitter;
+
+import twitter4j.GeoLocation;
+import twitter4j.Query;
+
+import java.io.*;
+import java.util.*;
+
+/**
+ * Stores and retrieves the maxID, oldestTweetID and the sinceID with the respective keyword/location for a search result which is used for the query generator
+ * during the next search
+ */
+public class IDHandler {
+    private static String directoryName = "Tweets_Search_Details";
+    private static String keywordMaxIDFilepath = directoryName + File.separator + "Keyword_MaxID1.txt";
+    private static String locationMaxIDFilepath = directoryName + File.separator + "Location_MaxID1.txt";
+
+    /**
+     * This method stores the maxID depending on the query
+     *
+     * @param maxID the lower bound for the further queries
+     * @param query determining the location based or keyword based
+     */
+    public void storeMaxID(double maxID, Query query) throws IOException {
+        //keywordBased MaxID
+        if (query.getQuery() != null) {
+            try (BufferedWriter out = new BufferedWriter(new FileWriter(new File(keywordMaxIDFilepath), true))) {
+                out.append(query.getQuery()).append(",").append(String.format("%.0f", maxID));
+            }
+        } else {
+            try (BufferedWriter out = new BufferedWriter(new FileWriter(new File(locationMaxIDFilepath), true))) {
+                out.append(String.valueOf(query.getGeocode())).append(",").append(String.format("%.0f", maxID));
+            }
+        }
+    }
+
+    /**
+     * This method searches the parameters stored which serves as the basis of query generation
+     *
+     * @param key for which the parameters have to be checked
+     * @return a list of the maxId, oldestTweetID and the sinceID as a list
+     */
+    public List<Long> keyParse(String key) throws IOException {
+        List<Long> list = new ArrayList<>();
+        list.add(-1L);
+        list.add(-1L);
+        list.add(-1L);
+        File file = new File(keywordMaxIDFilepath);
+        file.getParentFile().mkdirs();
+        file.createNewFile();
+        Scanner scan = new Scanner(file);
+        while (scan.hasNext()) {
+            String line = scan.nextLine();
+            if (line.contains(key)) {
+                String[] splitString = line.split(",");
+                list.set(0, Long.parseLong(splitString[1]));
+                list.set(1, Long.parseLong(splitString[2]));
+                list.set(2, Long.parseLong(splitString[3]));
```

Comment:
This seems to be error prone. If you take a look at the loop in which you are writing the maxID you can see that it is possible to end up with a line that contains only the maxID. However, your code assumes that always all three IDs are available.